src/lib.rs

```diff
@@ -984,6 +1032,35 @@ mod test {
     }
 
     #[test]
+    fn invalid_protocol() {
+        let creds = SchannelCredBuilder::new()
+            .enabled_protocols(&[Protocol::Ssl3])
+            .acquire(Direction::Outbound)
+            .unwrap();
+        let stream = TcpStream::connect("google.com:443").unwrap();
+        let stream = TlsStreamBuilder::new()
+            .domain("google.com")
+            .initialize(creds, stream);
+        assert_eq!(format!("{:?}", stream.err().unwrap()),
+                   "Error { repr: Custom(Custom { kind: Other, error: Error { code: 0x80090302, message: \"The function requested is not supported\" } }) }");
+
+        let creds = SchannelCredBuilder::new()
+            .enabled_protocols(&[Protocol::Tls12])
+            .acquire(Direction::Outbound);
+        assert_eq!(creds.err().unwrap().0, winapi::SEC_E_ALGORITHM_MISMATCH as winapi::DWORD);
+    }
+
+    #[test]
+    fn valid_protocol() {
+        // How to make this pass...
+        let creds = SchannelCredBuilder::new()
+            .enabled_protocols(&[Protocol::Tls10, Protocol::Tls11, Protocol::Tls12])
+            .supported_algorithms(&[Algorithm::RsaKeyx, Algorithm::Aes128, Algorithm::Sha1, Algorithm::Aes, Algorithm::Aes192, Algorithm::Aes256, Algorithm::Ecdh, Algorithm::RsaSign, Algorithm::Sha384, Algorithm::Sha512])
+            .acquire(Direction::Outbound);
+        assert_eq!(creds.err().unwrap().0, winapi::SEC_E_ALGORITHM_MISMATCH as winapi::DWORD);
```

Comment:
For some reason, I can't seem to get `enabled_algorithms` to *not* return a `SEC_E_ALGORITHM_MISMATCH`